### ISSUE TITLE
FEAT: Multiple tabs per table

### ISSUE THREAD

**Issue by carlosV2** [beekeeper-studio/beekeeper-studio#2096]
**Is your feature request related to a problem? Please describe.**
I can see on the [latest changelog](https://www.beekeeperstudio.io/blog/release-4.2-firebird-sql-files) that the maintainers gave up on having multiple tabs per table.
I'd like to show my use case and a potential solution to make everyone happy (hopefully).

More often than not I find myself filtering a table in search of an specific row that I need to update for reasons. However, due to my fish memory, I can't always remember the exact value I need to use so, before, I used to open another tab for the same table and check for the value or even run a second filtering until I find the value. 

**Describe the solution you'd like**
I can infer from the changelog that the "One tab per table" policy might have been massively requested so I won't attempt to change this. However, I'd like to suggest a new entry in the contextual menu for a table that reads "Open in new tab". This way, if anyone double clicks on a table, the system can check if a tab for that table is already open and take the user to that tab (which, by the way, currently, it clears any filter out of that tab 🤦🏼‍♂️) but if the user opens the context menu and selects "Open in new tab", then the system would open a new tab regardless if a tab for that table already exists.
If the user double clicks on a table with multiple tabs, then I'm happy for the community decision to decide on this. It could be to take the user to the first tab, to the last one, to the first open one, to the last open one, etc.

**Describe alternatives you've considered**
To be honest, the alternatives are pretty straight forward but equally annoying:
- Open a new window, connect to the database, open the table, filter it and port the values.
- Open the table, filter it, copy the value somewhere (i.e. clipboard if it is a single value or text file if they are many values), re-filter it and port the values.

**Additional context**
This is what happens when a table is filtered, another table is opened and I try to reopen the first table by double clicking on it's name in the list. As you can see, the filters bar has disappeared and the filter has been cleared (before it was showing a single record):
![image](https://github.com/beekeeper-studio/beekeeper-studio/assets/2642839/2440df71-74c4-4a33-a309-981cf714b788)

**Database type, if appropriate**
I am currently working with Postgresql databases so I can't test this on others but I'm sure the DB type isn't at play here.

**Comment by azmy60** [comment]
Hey @carlosV2 ,

There are many ways to handle this, and I think your solution is great. We just need to decide what's best for everyone. I would like to add that we're also working on beekeeper to be a little more configurable via config file. With that, another solution could be something like adding "multiple tabs per table" option in the config file, which is similar to VSCode.

**Comment by bediu** [comment]
@azmy60 

To solve this limit around one tab per table, I've resorted to opening multiple beekeeper studio instances.

I just think this reflects that this limit is just an inconvenience and doesn't really guard anything.

**Comment by not-night-but** [comment]
I think we could definitely add an 'Open in new tab' option. I think that is just the best of both worlds (I know this was a contentious issue for some people, so hopefully I'm right in that assessment lol)

**Comment by azmy60** [comment]
@not-night-but yeah, that would work. But be aware that we'll still have the issue with the foreign keys.

Let's say we use that to make 2 tabs of table A. What happens if we click a foreign key that points to table A?
